### PR TITLE
[SPARK-5339][BUILD] build/mvn doesn't work because of invalid URL for maven's tgz.

### PR DESCRIPTION
**Pull request by sarutak**
build/mvn will automatically download tarball of maven. But currently, the URL is invalid.

### CODE REVIEW

File: build/mvn [hunk 1]
```diff
@@ -68,7 +68,7 @@ install_app() {
 # Install maven under the build/ folder
 install_mvn() {
   install_app \
-    "http://apache.claz.org/maven/maven-3/3.2.3/binaries" \
+    "http://archive.apache.org/dist/maven/maven-3/3.2.3/binaries" \
```
Comment (srowen) [inline build/mvn]: While you're at it, point to 3.2.5? It's not ideal to point to `archive.apache.org` but at least it holds releases longer, whereas mirrors drop all but the latest point release in each line. I don't know of a redirector that would automatically send the request to a mirror though.

Comment (sarutak) [inline build/mvn]: Yeah, if there are any redirectors, I'd like to use it but I also don't know so I chose `archive.apache.org`. For now, it's reliable.

Comment (pwendell) [inline build/mvn]: @sarutak can you update to 3.2.5?

Comment (sarutak) [inline build/mvn]: Of course.

**Comment by sarutak** [comment]
@pwendell I've updated to `3.2.5` and confirmed build works with `build/mvn  package -DskipTests`.

**Comment by sarutak** [comment]
retest this please.

**Comment by pwendell** [comment]
Okay, LGTM good thanks!

**Pull request closed without merging**